wallet/src/de/schildbach/wallet/ui/main/WalletFragment.kt

```diff
@@ -93,6 +99,22 @@ class WalletFragment : Fragment(R.layout.home_content) {
         viewModel.transactions.observe(viewLifecycleOwner) { refreshShortcutBar() }
         viewModel.isBlockchainSynced.observe(viewLifecycleOwner) { updateSyncState() }
         viewModel.isBlockchainSyncFailed.observe(viewLifecycleOwner) { updateSyncState() }
+        viewModel.mostRecentTransaction.observe(viewLifecycleOwner) { mostRecentTransaction: Transaction ->
+            log.info("most recent transaction: {}", mostRecentTransaction.txId
+            )
+            if ((activity as? LockScreenActivity)?.lockScreenDisplayed != true && !configuration.hasDisplayedTaxCategoryExplainer
+                && WalletUtils.getTransactionDate(mostRecentTransaction).time >= configuration.taxCategoryInstallTime
+            ) {
+                val dialogFragment: TaxCategoryExplainerDialogFragment =
+                    TaxCategoryExplainerDialogFragment.newInstance(mostRecentTransaction.txId)
+                dialogFragment.show(activity?.supportFragmentManager!!, "taxcategorydialog") {
+                    val transactionDetailsDialogFragment: TransactionDetailsDialogFragment =
+                        TransactionDetailsDialogFragment.newInstance(mostRecentTransaction.txId)
+                    transactionDetailsDialogFragment.show(activity?.supportFragmentManager!!, null)
+                }
+                configuration.setHasDisplayedTaxCategoryExplainer()
+            }
+        }
```

Comment:
before the last merge, this code was in the WalletActivity class. 

In this class, we don't have direct access to LockScreenActivity.lockScreenDisplayed.  Our goal is not to display a dialog if the wallet is logged out.  

I don't like this way of fixing this problem.

Reply:
Yeah, sometimes fragments need to listen to or check the lock screen state too.
I'm thinking we should have a base class for the fragments for this case, something like `LockScreenAwareFragment` that will get notified by the `LockScreenActivity` the same way as it notifies (or rather dismisses) all its `DialogFragments`.